bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/discovery/LGThinqDiscoveryService.java

```diff
@@ -0,0 +1,149 @@
+import static org.openhab.core.thing.Thing.PROPERTY_MODEL_ID;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.lgthinq.internal.errors.LGThinqException;
+import org.openhab.binding.lgthinq.internal.handler.LGThinQBridgeHandler;
+import org.openhab.binding.lgthinq.lgservices.LGThinQApiClientServiceFactory;
+import org.openhab.binding.lgthinq.lgservices.LGThinQApiClientServiceFactory.LGThinQGeneralApiClientService;
+import org.openhab.binding.lgthinq.lgservices.model.LGDevice;
+import org.openhab.core.config.discovery.AbstractThingHandlerDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ServiceScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LGThinqDiscoveryService}
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+@Component(scope = ServiceScope.PROTOTYPE, service = LGThinqDiscoveryService.class)
+@NonNullByDefault
+public class LGThinqDiscoveryService extends AbstractThingHandlerDiscoveryService<LGThinQBridgeHandler> {
+
+    private final Logger logger = LoggerFactory.getLogger(LGThinqDiscoveryService.class);
+    private @Nullable ThingUID bridgeHandlerUID;
+    private @Nullable LGThinQGeneralApiClientService lgApiClientService;
+
+    public LGThinqDiscoveryService() {
+        super(LGThinQBridgeHandler.class, SUPPORTED_THING_TYPES, DISCOVERY_SEARCH_TIMEOUT);
+    }
+
+    @Override
+    public void initialize() {
+        bridgeHandlerUID = thingHandler.getThing().getUID();
+        // thingHandler is the LGThinQBridgeHandler
+        thingHandler.registerDiscoveryListener(this);
+        lgApiClientService = LGThinQApiClientServiceFactory
+                .newGeneralApiClientService(thingHandler.getHttpClientFactory());
+        super.initialize();
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Scan started");
+        // thingHandler is the LGThinQBridgeHandler
+        thingHandler.runDiscovery();
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan(), thingHandler.getThing().getUID());
+    }
+
+    public void removeLgDeviceDiscovery(LGDevice device) {
+        logger.debug("Thing removed from discovery: {}", device.getDeviceId());
+        try {
+            ThingUID thingUID = getThingUID(device);
+            thingRemoved(thingUID);
+        } catch (LGThinqException e) {
+            logger.error("Error getting Thing UID");
```

Comment:
Can you please adjust all logging in this binding to follow the coding guidelines. For reference:
https://www.openhab.org/docs/developer/guidelines.html#f-logging

Error => system wide error or binding specific bug. It is used alot in this binding, should be dropped by default. Only log if there is a benefit (check if not already logged by the framework / updateStatus) and at the right level (prefer debug and trace for troubleshooting)

```suggestion
            logger.warn("Error getting Thing UID");
```